sdks/java/core/src/main/java/org/apache/beam/sdk/util/common/Counter.java

```diff
@@ -1084,6 +1153,7 @@ public Integer getAggregate() {
 
     @Override
     public Counter<Integer> merge(Counter<Integer> that) {
+      setDirty();
```

Comment:
Dirtying should happen after any exception could be thrown, so the state of the Counter doesn't change if used improperly